src/server/service/kv.rs

```diff
@@ -911,41 +913,30 @@ impl<T: RaftStoreRouter + 'static> tikvpb_grpc::Tikv for Service<T> {
         stream: RequestStream<SnapshotChunk>,
         sink: ClientStreamingSink<Done>,
     ) {
-        let token = Token(self.token.fetch_add(1, Ordering::SeqCst));
-        let sched = self.snap_scheduler.clone();
-        let sched2 = sched.clone();
+        if self.snap_scheduler.task_count() >= self.max_snap_recv_concurrent {
```

Comment:
Is it better to control this inside the snapshot scheduler?

Reply:
We can't avoid an `AtomicUsize` because the downstream thread will pop every task very quickly and push them into CpuPool.